modules/fedimint-wallet-server/src/lib.rs

```diff
@@ -369,14 +381,21 @@ impl ServerModule for Wallet {
 
                 let new_consensus_block_count = self.consensus_block_count(dbtx).await;
 
-                // only sync from the first non-default consensus block count
-                if new_consensus_block_count > old_consensus_block_count {
-                    self.sync_up_to_consensus_height(
-                        dbtx,
-                        old_consensus_block_count,
-                        new_consensus_block_count,
-                    )
-                    .await;
+                debug!(
+                    ?peer_id,
+                    ?current_vote,
+                    ?block_count,
+                    ?old_consensus_block_count,
+                    ?new_consensus_block_count,
+                    "Received block count vote"
+                );
+
+                // only sync when we have a consensus block count
+                match (old_consensus_block_count, new_consensus_block_count) {
+                    (Some(old), Some(new)) if new > old => {
+                        self.sync_up_to_consensus_height(dbtx, old, new).await;
+                    }
+                    _ => {}
```

Comment:
We could assert a few things that should never happen (like going backwards) here. But I'll discuss that with @joschisan. Out of scope for this fix.